src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -304,7 +304,7 @@ export const createXMLHttpRequestOverride = (
           // Trigger a loadstart event to indicate the initialization of the fetch.
           this.trigger('loadstart')
 
-          this.status = mockedResponse.status || 200
+          this.status = mockedResponse.status ?? 200
```

Comment:
We need to add an integration test for mocking a status code that'd fall under this change. Mocking `0` status code may be a good example.